packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-swift-visitor.test.ts.snap

```diff
@@ -229,6 +465,229 @@ extension ModelPath where ModelType == Comment {
 }"
 `;
 
+exports[`AppSyncSwiftVisitor Primary Key Tests Should generate foreign key fields in hasMany uni relation for model with CPK customized foreign key names 1`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct Post: Model {
+  public let postId: String
+  public let title: String
+  public var comments: List<Comment>?
+  public var createdAt: Temporal.DateTime?
+  public var updatedAt: Temporal.DateTime?
+  
+  public init(postId: String,
+      title: String,
+      comments: List<Comment>? = []) {
+    self.init(postId: postId,
+      title: title,
+      comments: comments,
+      createdAt: nil,
+      updatedAt: nil)
+  }
+  internal init(postId: String,
+      title: String,
+      comments: List<Comment>? = [],
+      createdAt: Temporal.DateTime? = nil,
+      updatedAt: Temporal.DateTime? = nil) {
+      self.postId = postId
+      self.title = title
+      self.comments = comments
+      self.createdAt = createdAt
+      self.updatedAt = updatedAt
+  }
+}"
+`;
+
+exports[`AppSyncSwiftVisitor Primary Key Tests Should generate foreign key fields in hasMany uni relation for model with CPK customized foreign key names 2`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+extension Post {
+  // MARK: - CodingKeys 
+   public enum CodingKeys: String, ModelKey {
+    case postId
+    case title
+    case comments
+    case createdAt
+    case updatedAt
+  }
+  
+  public static let keys = CodingKeys.self
+  //  MARK: - ModelSchema 
+  
+  public static let schema = defineSchema { model in
+    let post = Post.keys
+    
+    model.pluralName = \\"Posts\\"
+    
+    model.attributes(
+      .index(fields: [\\"postId\\", \\"title\\"], name: nil),
+      .primaryKey(fields: [post.postId, post.title])
+    )
+    
+    model.fields(
+      .field(post.postId, is: .required, ofType: .string),
+      .field(post.title, is: .required, ofType: .string),
+      .hasMany(post.comments, is: .optional, ofType: Comment.self, associatedFields: [Comment.keys.postId]),
```

Comment:
This doesn't look accurate

Reply:
Opened issue to track this: https://github.com/aws-amplify/amplify-codegen/issues/539

Reply:
PR for the fix https://github.com/aws-amplify/amplify-codegen/pull/542